mobile/src/main/java/ch/epfl/sdp/mobile/ui/puzzles/PuzzleGameScreen.kt

```diff
@@ -5,39 +5,116 @@ import androidx.compose.foundation.rememberScrollState
+      }
   Scaffold(
       modifier = modifier,
+      scaffoldState = rememberScaffoldState(snackbarHostState = snackbarHostState),
+      topBar = {
+        GameScreenTopBar(
+            onBackClick = state::onBackClick,
+            onArClick = {},
+            onListenClick = state::onListenClick,
+            listening = state.listening,
+            modifier = Modifier.fillMaxWidth(),
+        )
+      },
       content = { scaffoldPadding ->
         Column(
             modifier =
                 modifier
                     .verticalScroll(rememberScrollState())
                     .padding(contentPadding)
                     .padding(scaffoldPadding)
-                    .padding(start = 16.dp, end = 16.dp, top = 48.dp, bottom = 48.dp),
-            verticalArrangement = Arrangement.spacedBy(32.dp),
+                    .padding(start = 32.dp, end = 32.dp, top = 48.dp, bottom = 48.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
         ) {
-          Text("Puzzle id: ${state.puzzle.uid}")
-          Text("Playing as: ${state.puzzle.playerColor}")
-          Text("Elo: ${state.puzzle.elo}")
+          Text(strings.puzzlesTitle.uppercase(), style = typo.h6)
+          PuzzleDirective(color = state.puzzleInfo.playerColor, puzzleState = state.puzzleState)
+          ProvideTextStyle(typo.subtitle1) { ClassicChessBoard(state) }
+          Spacer(Modifier)
+          Column {
+            Text(strings.puzzleNumber(state.puzzleInfo.uid), style = typo.subtitle2)
+            Text(strings.puzzleRating(state.puzzleInfo.elo.toString()), style = typo.subtitle2)
+          }
+          Moves(state.moves, Modifier.fillMaxWidth(), firstColor = firstColor)
         }
       },
   )
 }
+
+/**
+ * A composable which displays some basic information about what is going on in a puzzle.
+ *
+ * @param color the [ChessBoardState.Color] of the player.
+ * @param puzzleState the [PuzzleState] for this puzzle.
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+private fun PuzzleDirective(
+    color: ChessBoardState.Color,
+    puzzleState: PuzzleState,
+    modifier: Modifier = Modifier,
+) {
+  val strings = LocalLocalizedStrings.current
+  val colors = MaterialTheme.colors
+
+  @Composable
+  fun Directive() =
+      when (puzzleState) {
+        PuzzleState.Solving ->
+            Text(
+                strings.puzzleSolving(color.toString()),
+                color = colors.primary,
+                style = MaterialTheme.typography.subtitle1,
+            )
+        PuzzleState.Failed ->
+            Text(
+                strings.puzzleFailed,
+                color = colors.secondary,
+                style = MaterialTheme.typography.subtitle1,
+            )
+        PuzzleState.Solved ->
+            Text(
+                strings.puzzleSolved,
+                color = colors.primaryVariant,
+                style = MaterialTheme.typography.subtitle1,
+            )
+      }
```

Comment:
I'm maybe missing something, but what's the point of this function ? It looks like it's used only once, so inlining it will remove some lines of code.

(additionally, I'm not sure of the semantics of local composable functions, and when they are invalidated or what they capture)

Reply:
Fixed in https://github.com/epfl-SDP/android/pull/309/commits/69f53485392f4c208ade1179bce399a8838de369